app/decorators/current_user.rb

```diff
@@ -0,0 +1,46 @@
+class CurrentUser < Draper::Decorator
+  decorates :user
+  delegate_all
+
+  def session_url
+    if object
+      h.link_to("Logout",
+                h.logout_path,
+                method: :delete)
+    else
+      h.link_to("Login with GitHub",
+                "/auth/github",
+                method: :get)
+    end
+  end
+
+  def create_room_url
+    if object && !h.current_page?(h.new_room_path)
+      h.link_to("Create New Room",
+                h.new_room_path)
+    else
+      ""
+    end
+  end
+
+  def form_link(form)
+    if object
+      form.submit("Create room", class: "btn")
+    else
+      h.link_to("Login to create room",
+                "/auth/github",
+                class: "btn")
+    end
+  end
+
+  def close_topic(room)
+    if object == room.user
+      h.button_to("Close this Topic",
+                 h.room_path(room),
+                 method: :put,
+                 class: "btn col s12")
+    else
+      ""
+    end
+  end
+end
```

Comment:
I think conditionals in the view are neither conventional nor OO and that's the problem I'm trying to solve with this branch. I would use a PORO, but then I would have to pass the view context into that PORO anytime it needed to generate a link. I think getting a current_user functional early is good for the same reason that anything which reduces your technical debt is important early. Anytime we need a conditional in the view from here on it would be a single test in the decorator spec, a single method in the current user, and a single method in the view. I think that's so much better than something like ```<%= if current_user && @message.user == current_user || current_user.admin? || current_user.moderator? %>``` Which from the re-pivot I had A LOT of issues with.

Onboarding would be an issue, which makes me interested to see what Horace will say in our assessment because he worked on a giant product which probably had much more bewildering things new employees had to deal with.